config/src/test/java/org/springframework/security/config/annotation/web/configurers/oauth2/client/OAuth2LoginConfigurerTests.java

```diff
@@ -632,6 +621,42 @@ HttpSessionOAuth2AuthorizationRequestRepository oauth2AuthorizationRequestReposi
 		}
 	}
 
+	@Configuration
+	static class JwtDecoderFactoryConfig {
+
+		@Bean
+		JwtDecoderFactory<ClientRegistration> jwtDecoderFactory() {
+			return clientRegistration -> getJwtDecoder();
+		}
+
+		private static JwtDecoder getJwtDecoder() {
```

Comment:
I could return a `mock(JwtDecoder.class)` but mocking the return value of `decode()` will produce the same amount of code so doesn't really make it easier. However, it might make more sense to do it this way so will apply the change.